room.cpp

```diff
@@ -875,6 +877,69 @@ void Room::Private::updateDisplayname()
         emit q->displaynameChanged(q);
 }
 
+void Room::Private::toJson(QJsonObject &out) {
+    QJsonValue nowTimestamp { QDateTime::currentMSecsSinceEpoch() };
+    QJsonArray stateEvents;
+
+    QJsonObject nameEvent;
+    nameEvent["type"] = QString("m.room.name");
```

Comment:
One more thing: instead of `QString`, you'd want to use `QStringLiteral` in such cases.